src/DynamoPackages/PackageManagerClient.cs

```diff
@@ -277,9 +283,15 @@ internal PackageManagerResult Undeprecate(string name)
 
         internal bool DoesCurrentUserOwnPackage(Package package,string username) 
         {
+            bool value;
+            if (this.packageMaintainers.Count > 0 && this.packageMaintainers.TryGetValue(package.Name, out value)) {
+                return value;
+            }
             var pkg = new PackageInfo(package.Name, new Version(package.VersionName));
             var mnt = GetPackageMaintainers(pkg);
-            return (mnt != null) && (mnt.maintainers.Any(maintainer => maintainer.username.Equals(username)));
+            value = (mnt != null) && (mnt.maintainers.Any(maintainer => maintainer.username.Equals(username)));
```

Comment:
Just curious if `maintainers` could be null because it will crash Dynamo written this way

Reply:
`mnt.maintainers` cannot be null